zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/migration/MigrateCompensationSubscriptionTest.java

```diff
@@ -0,0 +1,520 @@
+                    .intermediateThrowEvent(
+                        "boundary_throw", ic -> ic.compensateEventDefinition().activityRef("B"))
+                    .endEvent()
+                    .done())
+            .deploy();
+    final long targetProcessDefinitionKey =
+        extractProcessDefinitionKeyByProcessId(deployment, targetProcessId);
+
+    final var processInstanceKey = engine.processInstance().ofBpmnProcessId(processId).create();
+
+    RecordingExporter.jobRecords(JobIntent.CREATED)
+        .withProcessInstanceKey(processInstanceKey)
+        .withElementId("A")
+        .await();
+
+    // when
+    engine
+        .processInstance()
+        .withInstanceKey(processInstanceKey)
+        .migration()
+        .withTargetProcessDefinitionKey(targetProcessDefinitionKey)
+        .addMappingInstruction("A", "B")
+        .migrate();
+
+    // then
+    Assertions.assertThat(
+            RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_MIGRATED)
+                .withProcessInstanceKey(processInstanceKey)
+                .withElementType(BpmnElementType.PROCESS)
+                .getFirst())
+        .describedAs("Expect that the process instance is migrated")
+        .isNotNull();
+
+    engine.job().ofInstance(processInstanceKey).withType("A").complete();
+
+    Assertions.assertThat(
+            RecordingExporter.compensationSubscriptionRecords()
+                .withIntent(CompensationSubscriptionIntent.CREATED)
+                .withProcessInstanceKey(processInstanceKey)
+                .getFirst()
+                .getValue())
+        .describedAs("Expect that the process definition is updated")
+        .hasProcessDefinitionKey(targetProcessDefinitionKey)
+        .describedAs("Expect that the compensable activity is updated")
+        .hasCompensableActivityId("B")
+        .describedAs("Expect that the compensation handler id is unchanged")
+        .hasCompensationHandlerId("undoB")
+        .describedAs("Expect that the compensation boundary event in the source is created")
+        .isNotNull();
+
+    engine.job().ofInstance(processInstanceKey).withType("undoB").complete();
+
+    Assertions.assertThat(
+            RecordingExporter.compensationSubscriptionRecords()
+                .withIntent(CompensationSubscriptionIntent.COMPLETED)
+                .withProcessInstanceKey(processInstanceKey)
+                .getFirst()
+                .getValue())
+        .describedAs("Expect that the compensation boundary event in the target is completed")
+        .hasProcessDefinitionKey(targetProcessDefinitionKey)
+        .describedAs("Expect that the compensable activity is updated")
+        .hasCompensableActivityId("B")
+        .describedAs("Expect that the compensation handler id is unchanged")
+        .hasCompensationHandlerId("undoB")
+        .describedAs("Expect that the compensation boundary event in the source is created")
+        .isNotNull();
```

Comment:
🔧 I'll move this at the beginning of the chain

Reply:
It is unnecessary, completely removed it.